docs/ops/convolution/BinaryConvolution_1.md

```diff
@@ -27,13 +62,120 @@ The operation has the same attributes as a regular *Convolution* layer and sever
   * **Default value**: None
   * **Required**: *yes*
 
-**Inputs**:
+* *auto_pad*
 
-*   **1**: ND tensor with N >= 3, containing integer, float or binary values; filled with 0/1 values of any appropriate type. 0 means -1, 1 means 1 for `mode="xnor-popcount"`. Required.
+  * **Description**: *auto_pad* how the padding is calculated. Possible values:
+    * *explicit* - use explicit padding values from *pads_begin* and *pads_end*.
+    * *same_upper* - the input is padded to match the output size. In case of odd padding value an extra padding is added at the end.
+    * *same_lower* - the input is padded to match the output size. In case of odd padding value an extra padding is added at the beginning.
+    * *valid* - do not use padding.
+  * **Type**: string
+  * **Default value**: explicit
+  * **Required**: *no*
+  * **Note**: *pads_begin* and *pads_end* attributes are ignored when *auto_pad* is specified.
+  
+**Inputs**:
 
-*   **2**: ND tensor with N >= 3 that represents convolutional kernel filled by integer, float or binary values; filled with 0/1 values. 0 means -1, 1 means 1 for `mode="xnor-popcount"`. Required.
+*   **1**: Input tensor of type *T* and rank 3, 4 or 5. Layout is NCZYX (number of batches, number of channels, spatial axes Z, Y, X). Required.
+*   **2**: Kernel tensor of type *T* and rank 3, 4 or 5. Layout is OIZYX (number of output channels, number of input channels, spatial axes Z, Y, X). Required.
+*   **Note**: Interpretation of tensor values is defined by *mode* attribute.
+*   **Note**: Type of the convolution (1D, 2D or 3D) is derived from the rank of the input tensors and not specified by any attribute:
+      * 1D convolution (input tensors rank 3) means that there is only one spatial axis X
+      * 2D convolution (input tensors rank 4) means that there are two spatial axes Y, X
+      * 3D convolution (input tensors rank 5) means that there are three spatial axes Z, Y, X
 
 **Outputs**:
 
-*   **1**: output tensor containing float values.
+*   **1**: Output tensor of type *T* and rank 3, 4 or 5. Layout is NOZYX (number of batches, number of kernel output channels, spatial axes Z, Y, X).
+  
+**Types**:
+
+* *T*: any floating point type with values `0` or `1`.
```

Comment:
I have doubts about the input/output types. Still investigating how it looks in CPU plugin...

Reply:
From [this](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_bin_conv_node.cpp#L986) code it looks like we are expected to have both inputs of binary type and output as fp32. @lazarevevgeny , @dmitry-gorokhov can you comment on this? I'm trying to establish which types should be used for input and output tensor in BinaryConvolution.

Reply:
@iefode did you manage to successfully run any test with binary convolution at all? Together with @ggalieroc we tried different combinations of input types and CPU plugin always throws an exception. Can you provide correct types or confirm that plugin is not working at all? I'd like to have the simplest working unit test with inference, like [here](https://github.com/openvinotoolkit/openvino/compare/master...jdanieck:binary_convolution).

Reply:
@yury-intel maybe you can advise here? Which input/output tensor types should be supported by CPU plugin?

Reply:
Based on IR bellow I've established that supported types should be: 
- FP32 for data
- BIN for kernel
- FP32 for output

But test still fails ... 

**IR: Intel_Resnet18_xnor_Binary_onnx_0001_IR_v10_FP32-INT1_fused_batch_1**
```
<layer id="15" name="data" type="FakeQuantize" version="opset1">
	<output> 
		<port id="5" precision="FP32"></port>
	</output>
</layer>

<layer id="16" name="kernel" type="Const" version="opset1">
	<data element_type="u1" />
	<output> 
		<port id="0" precision="BIN"></port> 
	</output>
</layer>

<layer id="17" name="Conv" type="BinaryConvolution" version="opset1">
	<input>
		<port id="0"></port>
		<port id="1"></port>
	</input>
		<output>
			<port id="2" precision="FP32"></port>
		</output>
</layer>

<edge from-layer="15" from-port="5" to-layer="17" to-port="0" />
<edge from-layer="16" from-port="0" to-layer="17" to-port="1" />
<edge from-layer="17" from-port="2" to-layer="22" to-port="0" />
```

**UnitTest:**
```
inputs_param = make_shared<op::Parameter>(element::f32, inputs_shape);
auto filters_param = make_shared<op::Parameter>(element::u1, filter_shape);
auto conv = make_shared<op::v1::BinaryConvolution>(...);
auto f = make_shared<Function>(conv, ParameterVector{inputs_param, filters_param});
```

**Test execution:**
```
C++ exception with description "[NOT_IMPLEMENTED] Input image format BIN is not supported yet...
../inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:335
../inference-engine/include/details/ie_exception_conversion.hpp:64" thrown in the test body.
```

Reply:
@jdanieck Looks like weights `auto filters_param = make_shared<op::Parameter>(element::u1, filter_shape);` should be const - not parameter.

Reply:
@dmitry-gorokhov thanks for reply. Can you explain why it needs to be Constant? For other Convolutions there was no such requirement. Either way [u1 type is unsupported](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/ngraph/op/constant.hpp#L579) in Constant., so I can't create one with this type. That's why I don't understand how this is supposed to be working and until we establish for which input/output types BinaryConvolution should work we can't really progress with reference implementation.  I'd really appreciate your help here, we are not long in this project and it takes a lot of time to reverse engineer this stuff.

Reply:
@jdanieck Requirement to be constant is the CPU plug-in limitation. From the operation specification standpoint it can be parameter as well, but we don't support it since overall binary networks support in OV is very limited.

As for u1 type Constant node creation we have an example: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/convert_u1_to_string.cpp
Also adding @ilyachur since he can comment better here.

Reply:
Ok. I've been able to write working unit test. I'll go from there by myself. Thank you for your time!